File: 004_InsertNodEndLinkedList/test040.cpp
/**
 * @file test040.cpp
 * @author M. Z.
 * @brief Inserting a node at the end of a linked list
 * @version 0.1
 * @date 2025-03-03
 *
 * @copyright Copyright (c) 2025
 *
 */
#include <iostream>

using namespace std;

class Node {
public:
  int data;
  Node *next;

  Node(int data) {
    this->data = data;
    this->next = nullptr;
  }
};

class LinkedList {
public:
  Node *head;
  
  LinkedList() { this->head = nullptr; }

  void insertNodeAtEndOfList(int data) {
    Node *temp1 = new Node(data);
    temp1->data = data;
    temp1->next= nullptr;
    if (head == nullptr) {
      head = temp1;
    }else{
    Node *temp2 = head;
    while (temp2->next != nullptr) { // (position = 3) means the first swap
      temp2 = temp2->next;
    }
    temp2->next = temp1;      
  }
  }

  void printList() {
    Node *temp = head;
    cout << "List is: ";
    while (temp != NULL) {
      cout << " " << temp->data;
      temp = temp->next;
    }
    cout << endl;
  }
};

int main() {
  LinkedList LL;
  LL.insertNodeAtEndOfList(2);
  LL.insertNodeAtEndOfList(3);
  LL.insertNodeAtEndOfList(4);
  LL.insertNodeAtEndOfList(5);
  LL.printList();
}

File: 004_InsertNodEndLinkedList/test040.py
'''
 * @file test040.py
 * @author M. Z.
 * @brief Inserting a node at the end of a linked list
 * array
 * @date 2025-03-03
 *
 * @copyright Copyright (c) 2025
'''
import sys

def Node(data, next):
    return [data, next]

def insertNodeAtEndOfList(head, data):
    temp1 = Node(data, None) #temp1[0] = data, temp1[1] = next
    if (head == None):
        head = temp1
    else:
        temp2 = head
        while (temp2[1] != None):
           temp2 = temp2[1]
        temp2[1] = temp1
    return head

def printList():
  temp = head
  print("The list is: ", end=" ")
  while temp != None:
    print(temp[0], end=" ")
    temp = temp[1]
  print() #newline
    
head = None  # Empty linked list
head = insertNodeAtEndOfList(head, 9)
head = insertNodeAtEndOfList(head, 8)
head = insertNodeAtEndOfList(head, 7)
head = insertNodeAtEndOfList(head, 6)
head = insertNodeAtEndOfList(head, 5)
printList()


